File: code/Ch6.2/bigbob10.c
/* Simple C client example.
 * Based on simple.c from player distribution
 * K. Nickels 6/8/14
 */

#include <stdio.h>
#include <libplayerc/playerc.h>

int main(int argc, char *argv[]) {
  playerc_client_t *robot;
  playerc_blobfinder_t *blobfinderProxy;
  playerc_blobfinder_blob_t blob;

  /* Create a client and connect it to the server. */
  robot = playerc_client_create(NULL, "localhost", 6665);
  if (0 != playerc_client_connect(robot)) return -1;

  /* Create and subscribe to a blobfinder device. */
  blobfinderProxy = playerc_blobfinder_create(robot, 0);
  if (playerc_blobfinder_subscribe(blobfinderProxy, PLAYER_OPEN_MODE)) return -1;

  /* read from the proxies */
  playerc_client_read(robot);

  /* print out stuff for fun */
  printf("Blobfinder, width=%d, height=%d\n",
		  blobfinderProxy->width,
		  blobfinderProxy->height);
  printf("%d blobs found\n",
		  blobfinderProxy->blobs_count);

  for (int i=0;i<blobfinderProxy->blobs_count;i++) {
	  blob = blobfinderProxy->blobs[i];
	  printf("BLOB %d -----------\n", i);
	  printf("-id=%d\n",blob.id);
	  printf("-color=0x%06x\n",blob.color);
	  printf("-area=%d (pixels)\n",(int)blob.area);
	  printf("-range=%d (m)\n",(int)blob.range);
	  printf("-(x,y) = (%d,%d)\n",blob.x,blob.y);
	  printf("-box =   %3d\n",blob.top);
	  printf("-     %3d   %3d\n",blob.left,blob.right);
	  printf("-        %3d\n",blob.bottom);
  }

  /* Shutdown */
  playerc_blobfinder_unsubscribe(blobfinderProxy);
  playerc_blobfinder_destroy(blobfinderProxy);
  playerc_client_disconnect(robot);
  playerc_client_destroy(robot);

  return 0;
}

File: code/Ch7.2/bigbob10.py
# Simple python client example 
# Based on example0.cc from player distribution
# K. Nickels 7/24/13

import math, sys, os
sys.path.append('/usr/local/lib64/python2.7/site-packages/')
from playercpp import *

# Make proxies for Client, blobfinder
robot = PlayerClient("localhost");
bf = BlobfinderProxy(robot,0);

# Read from the proxies
robot.Read();


# Print stuff out for fun
for i in range(bf.GetCount()):
	blob = bf.GetBlob(i);
	print 'BLOB %d, ' % i
        # Can't seem to access .color .x .y .top .left .right .bottom from
        # python.  
		# blob is a pointer to a C structure playerc_blob_t
		# Probably a SWIG problem...

	robot.Read()
